### PR TITLE
Use a game action result for clear at function

### PR DESCRIPTION
**Pull request by duncanspumpkin**
I've been meaning to do this for a while.
The idea is to make it so that the error message args and ground flags are no longer passed around in global variables. Note this does not work at the moment as I need to modify the other callers of `map_obstruction_set_error_text`.

Argh can't believe I finally did what I always tell people not to do and recreated a pull request.

### CODE REVIEW

File: src/openrct2/actions/WallPlaceAction.hpp [hunk 1]
```diff
@@ -380,16 +380,16 @@ DEFINE_GAME_ACTION(WallPlaceAction, GAME_COMMAND_PLACE_WALL, WallPlaceActionResu
         bool wallAcrossTrack = false;
         if (!(GetFlags() & GAME_COMMAND_FLAG_PATH_SCENERY) && !gCheatsDisableClearanceChecks)
         {
-            if (!WallCheckObstruction(wallEntry, targetHeight / 8, clearanceHeight, &wallAcrossTrack))
+            if (auto result = WallCheckObstruction(wallEntry, targetHeight / 8, clearanceHeight, &wallAcrossTrack);
```
Comment (Gymnasiast) [inline src/openrct2/actions/WallPlaceAction.hpp]: I think it's better not to stuff the assignment into the `if` - it makes for harder-to-read code IMO.

File: src/openrct2/actions/WallPlaceAction.hpp [hunk 2]
```diff
@@ -603,18 +603,16 @@ DEFINE_GAME_ACTION(WallPlaceAction, GAME_COMMAND_PLACE_WALL, WallPlaceActionResu
      *
      *  rct2: 0x006E5C1A
      */
-    bool WallCheckObstruction(rct_scenery_entry * wall, int32_t z0, int32_t z1, bool* wallAcrossTrack) const
+    GameActionResult::Ptr WallCheckObstruction(rct_scenery_entry * wall, int32_t z0, int32_t z1, bool* wallAcrossTrack) const
     {
         int32_t entryType, sequence;
         rct_scenery_entry* entry;
         rct_large_scenery_tile* tile;
-
```
Comment (tupaschoal) [inline src/openrct2/actions/WallPlaceAction.hpp]: Seems like unrelated change
